File: Map/Map/Map/statement/OpenRFile.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Map.domain;
using System.IO;

namespace Map.statement
{
    public class OpenRFile : IStatement
    {
        private String varFile, filePath;
        private String fileName;

        public OpenRFile(String varf, String path)
        {
            varFile = varf;
            filePath = path;
            String[] file = filePath.Split('\\');
            fileName = file[file.Length - 1];
        }

        private bool exists(ProgramState ps, String path)
        {
            foreach (int k in ps.fileTable.getAll())
                if (ps.fileTable.getValue(k).filePath.Equals(path))
                    return true;
            return false;
        }

        public ProgramState execute(ProgramState ps)
        {
            if (exists(ps, filePath))
                throw new Exception("The file " + filePath + " is already opened!");
            try
            {
                StreamReader sr = File.OpenText(filePath);

                FileData fd = new FileData(filePath, sr);
                int id = FileIDGenerator.generateId();
                ps.fileTable.add(id, fd);
                if (ps.symbolTable.contains(varFile))
                    ps.symbolTable.update(varFile, id);
                else
                    ps.symbolTable.add(varFile, id);
            }
            catch(IOException e)
            {
                throw new Exception(e.Message);
            }
            return null;
        }

        public override string ToString()
        {
            return "Openfile(" + varFile + "," + fileName + ") ";
        }
    }
}

File: Map/Z_InterpretorWCDL/src/statement/OpenRFile.java
package statement;

import java.io.BufferedReader;
import java.io.FileReader;
import java.io.IOException;

import domain.DomainException;
import domain.FileData;
import domain.FileIDGenerator;
import domain.IFileTable;
import domain.PrgState;

public class OpenRFile implements IStatement 
{
	
	private String varFile;
	private String fileName;
	
	public OpenRFile(String varf, String fn)
	{
		varFile = varf;
		fileName = fn;
	}
	
	private boolean exists(PrgState p, String fileName)
	{
		IFileTable<Integer,FileData> ft = p.getFileTable();
		for(Integer key : ft.getAll())
		{
			if(ft.getValue(key).getFileName().equals(fileName))
				return true;
		}
		return false;
	}
	
	@Override
	public PrgState execute(PrgState ps) 
	{
		if(exists(ps,fileName))
			throw new DomainException("The file: "+fileName+" is already open!");
		try
		{
			BufferedReader br = new BufferedReader(new FileReader(fileName));
			
			FileData data = new FileData(fileName,br);
			int id = FileIDGenerator.generateID();
			ps.getFileTable().add(id,data);
			if(ps.getSymbolTable().contains(varFile))
				ps.getSymbolTable().update(varFile,id);
			else
				ps.getSymbolTable().add(varFile,id);
			
		} 
		catch (IOException e) 
		{
			throw new DomainException(e.getMessage());
		}
		return null;
	}
	
	@Override
	public String toString()
	{
		return "Openfile("+varFile+","+fileName+") ";
	}

}
